In [1]:
# using this to grab the other 3 data sets

In [2]:
import pandas as pd

In [3]:
df_beers = pd.read_csv('../../data/beers.csv')
df_breweries = pd.read_csv('../../data/breweries.csv')
df_reviews = pd.read_csv('../../data/reviews.csv')


In [4]:
print(df_beers.info())
print(df_beers.isna().sum())
df_beers.head()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 358873 entries, 0 to 358872
Data columns (total 10 columns):
 #   Column        Non-Null Count   Dtype  
---  ------        --------------   -----  
 0   id            358873 non-null  int64  
 1   name          358873 non-null  object 
 2   brewery_id    358873 non-null  int64  
 3   state         298147 non-null  object 
 4   country       358719 non-null  object 
 5   style         358872 non-null  object 
 6   availability  358873 non-null  object 
 7   abv           320076 non-null  float64
 8   notes         358827 non-null  object 
 9   retired       358873 non-null  object 
dtypes: float64(1), int64(2), object(7)
memory usage: 27.4+ MB
None
id                  0
name                0
brewery_id          0
state           60726
country           154
style               1
availability        0
abv             38797
notes              46
retired             0
dtype: int64


id                                  name  brewery_id state country  \
0  202522                        Olde Cogitator        2199    CA      US   
1   82352  Konrads Stout Russian Imperial Stout       18604   NaN      NO   
2  214879                        Scottish Right       44306    IN      US   
3  320009               MegaMeow Imperial Stout        4378    WA      US   
4  246438                       Peaches-N-Cream       44617    PA      US   

                     style availability   abv                   notes retired  
0    English Oatmeal Stout     Rotating   7.3  No notes at this time.       f  
1   Russian Imperial Stout     Rotating  10.4  No notes at this time.       f  
2             Scottish Ale   Year-round   4.0  No notes at this time.       t  
3  American Imperial Stout       Winter   8.7    Every time this year       f  
4       American Cream Ale     Rotating   5.1  No notes at this time.       f

Looking at the beers dataframe, there's a few things I'm noticing:

1. The sheer number of missing states seems a bit daunting but one thing that is very identifiable in the head print out above is that those beers seem to international beers where there isn't a state concept similar to the US. I haven't checked this as a foolproof method however I'd be willing to bet those beers are brewed outside the US.
2. I'll have to look into the missing 154 country values but might just drop those given how insignificant those beers might be to the ~360k data points we have in this set.
3. I'm going to replace those NaN notes to just be "No notes at this time." so that it's similar to other tasting notes.

In [11]:
df_beers.dropna(subset=['country', 'style'], inplace=True)

In [15]:
df_beers['notes'].fillna('No notes at this time', inplace=True)
df_beers.isna().sum()

id                  0
name                0
brewery_id          0
state           60572
country             0
style               0
availability        0
abv             38787
notes               0
retired             0
dtype: int64

In [26]:
df_beers['state'].fillna('Not Applicable', inplace=True)

In [27]:
df_beers.isna().sum()

id                  0
name                0
brewery_id          0
state               0
country             0
style               0
availability        0
abv             38787
notes               0
retired             0
dtype: int64

In [28]:
df_beers.drop(columns='abv', inplace=True, axis=1)

In [29]:
df_beers.isna().sum()

id              0
name            0
brewery_id      0
state           0
country         0
style           0
availability    0
notes           0
retired         0
dtype: int64

In [7]:
print(df_breweries.info())
print(df_breweries.isna().sum())
df_breweries.head()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 50347 entries, 0 to 50346
Data columns (total 7 columns):
 #   Column   Non-Null Count  Dtype 
---  ------   --------------  ----- 
 0   id       50347 non-null  int64 
 1   name     50347 non-null  object
 2   city     50289 non-null  object
 3   state    39076 non-null  object
 4   country  50341 non-null  object
 5   notes    50262 non-null  object
 6   types    50347 non-null  object
dtypes: int64(1), object(6)
memory usage: 2.7+ MB
None
id             0
name           0
city          58
state      11271
country        6
notes         85
types          0
dtype: int64


id                                  name            city state country  \
0  19730                       Brouwerij Danny       Erpe-Mere   NaN      BE   
1  32541           Coachella Valley Brewing Co  Thousand Palms    CA      US   
2  44736                      Beef 'O' Brady's      Plant City    FL      US   
3  23372                Broadway Wine Merchant   Oklahoma City    OK      US   
4  35328  Brighton Beer Dispensary (DUPLICATE)        Brighton   GB2      GB   

                                               notes                     types  
0                             No notes at this time.                   Brewery  
1                             No notes at this time.  Brewery, Bar, Beer-to-go  
2                             No notes at this time.               Bar, Eatery  
3                             No notes at this time.                     Store  
4  Duplicate of https://www.beeradvocate.com/beer...               Bar, Eatery

A lot of this data already exists in the other tables, not going to use this table at all. Will keep it under the variable however will not look to join these tables together.

In [8]:
print(df_reviews.info())
print(df_reviews.isna().sum())
df_reviews

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 9073128 entries, 0 to 9073127
Data columns (total 10 columns):
 #   Column    Dtype  
---  ------    -----  
 0   beer_id   int64  
 1   username  object 
 2   date      object 
 3   text      object 
 4   look      float64
 5   smell     float64
 6   taste     float64
 7   feel      float64
 8   overall   float64
 9   score     float64
dtypes: float64(6), int64(1), object(3)
memory usage: 692.2+ MB
None
beer_id           0
username       3815
date              0
text              0
look        3790018
smell       3790018
taste       3790018
feel        3790018
overall     3790018
score             0
dtype: int64


beer_id          username        date  \
0         271781      bluejacket74  2017-03-17   
1         125646           _dirty_  2017-12-21   
2         125646           CJDUBYA  2017-12-21   
3         125646   GratefulBeerGuy  2017-12-20   
4         125646          LukeGude  2017-12-20   
...          ...               ...         ...   
9073123   104824            CTJman  2014-06-08   
9073124   104824  IMSPEAKNOENGLISH  2014-06-06   
9073125   104824          twizzard  2014-06-05   
9073126   104824        bootdown21  2014-06-04   
9073127   104824           Brensol  2014-06-01   

                                                      text  look  smell  \
0           750 ml bottle, 2016 vintage, bottle #304 of...  4.00   4.00   
1                                                           4.50   4.50   
2                                                           4.75   4.75   
3           0% 16 oz can. Funny story: As I finally wal...  4.75   4.75   
4           Classic TH NEIPA. Overflowing head and bouq...  4.25   4.50   
...                                                    ...   ...    ...   
9073123                                                      NaN    NaN   
9073124                                                      NaN    NaN   
9073125                                                      NaN    NaN   
9073126                                                      NaN    NaN   
9073127                                                      NaN    NaN   

         taste  feel  overall  score  
0         4.00  4.25     4.00   4.03  
1         4.50  4.50     4.50   4.50  
2         4.75  4.75     4.75   4.75  
3         4.50  4.50     4.50   4.58  
4         4.25  4.25     4.25   4.31  
...        ...   ...      ...    ...  
9073123    NaN   NaN      NaN   4.00  
9073124    NaN   NaN      NaN   5.00  
9073125    NaN   NaN      NaN   5.00  
9073126    NaN   NaN      NaN   4.50  
9073127    NaN   NaN      NaN   4.75  

[9073128 rows x 10 columns]

This dataset is massive. I'm just going to drop all values for now and then look at these if I end up having a better model or idea of how I want to use my data, I'll come back and revisit this.

In [30]:
df_reviews.dropna(inplace=True)

In [31]:
df_reviews.isna().sum()

beer_id     0
username    0
date        0
text        0
look        0
smell       0
taste       0
feel        0
overall     0
score       0
dtype: int64